### PR TITLE
Fix .c, .fml

### PR DESCRIPTION
**Pull request by paulwalko**
.c now scrapes from google search at least until ddg fixes their api, and a generic google search function was added that works for searching for search results and calc results

mylife changed their html again

### CODE REVIEW

**Comment by paulwalko** [comment]
I did something wrong...

**Comment by paulwalko** [comment]
Why is github including all the older commits as well as the most recent 4? I thought those were already merged previously.

**Comment by paulwalko** [comment]
Nevermind... I'm dumb

File: modules/calc.py [hunk 1]
```diff
@@ -20,30 +21,55 @@
...
-    if answer:
-        phenny.say(answer)
+    if num:
+        phenny.say(num)
     else:
-        phenny.reply('Sorry, no result.')
+        phenny.reply("Sorry, no result.")
+
+
+# def c(phenny, input):
```
Comment (mutantmonkey) [inline modules/calc.py]: Please remove this old code. It's no problem to pull stuff from the Git history if it turns out that we need it later.

File: modules/test/test_calc.py [hunk 2]
```diff
@@ -29,7 +29,7 @@ def test_c_scientific(self):
         input = Mock(group=lambda x: '2^64')
         c(self.phenny, input)
 
-        self.phenny.say.assert_called_once_with('1.84467440737096 * 10^19')
+        self.phenny.say.assert_called_once_with('1.84467441 × 10^19')
```
Comment (mutantmonkey) [inline modules/test/test_calc.py]: I don't think it's reasonable to expect people to use Unicode here. Is there really no other way to do this?

File: modules/weather.py [hunk 3]
```diff
@@ -17,8 +17,14 @@
...
 def location(q):
-    uri = 'https://nominatim.openstreetmap.org/search/?q={query}&format=json'.\
-        format(query=web.quote(q))
+    uri = 'https://nominatim.openstreetmap.org/search?%s={query}&format=json'
+    if q.isdigit():
+        uri = uri % 'postalcode'
+    else:
+        uri = uri % 'q'
+    uri = uri . format(query = web.quote(q))
+#    uri = 'https://nominatim.openstreetmap.org/search/?q={query}&format=json'.\
```
Comment (mutantmonkey) [inline modules/weather.py]: As with the other change, please remove unused code.

File: modules/weather.py [hunk 4]
```diff
@@ -17,8 +17,14 @@
 
 
 def location(q):
-    uri = 'https://nominatim.openstreetmap.org/search/?q={query}&format=json'.\
-        format(query=web.quote(q))
+    uri = 'https://nominatim.openstreetmap.org/search?%s={query}&format=json'
```
Comment (mutantmonkey) [inline modules/weather.py]: I like this change, but can you please use `format` instead of the the old `%` way?

**Comment by mutantmonkey** [comment]
Thanks!

**Pull request merged**